pkg/crawler/crawler.go

```diff
@@ -586,24 +594,72 @@ func (c *Crawler) generateLicenseFile(client http.Client, licenseFileName string
 
 	}
 
-	// download license url contents
+	// get license url contents
 	resp, err := client.Get(licenseMeta.URL)
-	if resp == nil {
-		return false, nil
-	}
-
-	if resp.StatusCode == http.StatusNotFound {
+	if err != nil {
+		log.Printf("Error while fetching license Meta URL, error %s", err.Error())
 		return false, nil
 	}
-	if err != nil {
+	if resp.StatusCode != http.StatusOK {
+		log.Printf("Error while fetching license Meta URL %s, status code %d", licenseMeta.URL, resp.StatusCode)
 		return false, nil
 	}
 	defer resp.Body.Close()
 
-	_, err = io.Copy(f, resp.Body)
+	responseBody, err := handleCompressedResponsebody(resp)
 	if err != nil {
 		return false, nil
 	}
+	os.WriteFile(licenseFileName, []byte(responseBody), 0644)
 
 	return true, nil
 }
+
+func handleCompressedResponsebody(resp *http.Response) (string, error) {
+	var reader io.Reader = resp.Body
+	switch resp.Header.Get("Content-Encoding") {
+	case "gzip":
+		gzipReader, err := gzip.NewReader(reader)
+		if err != nil {
+			return "", err
+		}
+		defer gzipReader.Close()
+		reader = gzipReader
+
+	case "br":
+		reader = brotli.NewReader(reader)
+
+	case "deflate":
+		deflateReader := flate.NewReader(reader)
+		defer deflateReader.Close()
+		reader = deflateReader
+	}
+
+	body, err := io.ReadAll(reader)
```

Comment:
io.Copy is preferred for large files if i am not wrong, hence we were using io.Copy to a file earlier. It transfers data in parts.

Reply:
got it. added it back